examples/pytorch/fsdp/fine-tune-t5-with-fsdp.ipynb

```diff
@@ -0,0 +1,509 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Fine-Tune T5 Model with PyTorchJob and FSDP"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This Notebook will fine-tune Text-to-Text Transfer Transformer (T5) with Wikihow dataset for text summarization using Kubeflow PyTorchJob.\n",
+    "\n",
+    "Pretrained T5 model: https://huggingface.co/google-t5/t5-base\n",
+    "\n",
+    "Wikihow dataset: https://github.com/mahnazkoupaee/WikiHow-Dataset\n",
+    "\n",
+    "This Notebook will use **4** GPUs to fine-tune T5 model on 2 Nodes. This example is based on [the official PyTorch FSDP tutorial](https://pytorch.org/tutorials/intermediate/FSDP_adavnced_tutorial.html)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## FSDP with multi-node multi-worker training"
```

Comment:
[the official PyTorch FSDP tutorial](https://pytorch.org/tutorials/intermediate/FSDP_adavnced_tutorial.html) showcase the training on a single node.

In this example, multi-node is used for training, and the model is not saved, correct?
If the model needs to be saved, is it necessary to use PVC and S3?

Reply:
Similar to this example, we can save model on the process with RANK=0:
https://pytorch.org/tutorials/intermediate/FSDP_adavnced_tutorial.html#:~:text=if%20rank%20%3D%3D%200%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20print(f%22%2D%2D%3E%20saving,)%0A%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20torch.save(cpu_state%2C%20save_name)